### PR TITLE
[receiver/prometheusreceiver] Configuring httpSD http client from TAs clienthttp

### LINKED ISSUE #33370: TLS config in TA receiver doesn't take effect
### Component(s)

receiver/prometheus

## Description

After https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31449 was merged it is expected that the client will use the TLS setting to communicate with the TA.

When adding TLS config to the config.yaml file, the client is not sending the tls certs in the requests to the Target Allocator.

In practice the TA server (or mock of) complains that the client does not provide certificates in the requests.

## Steps to Reproduce

Set up a TA (Or mock) with mTLS - forcing the client to provide certificates.

Setup the collector with the tls key in the config:

collector.yaml
```yaml
receivers:
    prometheus:
        config: {}
        target_allocator:
            collector_id: ${POD_NAME}
            endpoint: https://tls-debug-service:443
            interval: 30s
            tls:
                ca_file: /tls/ca.crt
                cert_file: /tls/tls.crt
                key_file: /tls/tls.key
```

## Expected Result

TLS certificates added to the request

## Actual Result

Server complains that the client didn't provide a certificate

From the TA (or mock):
TA:
```
http: TLS handshake error from 10.42.0.201:40804: tls: client didn't provide a certificate
```

From mock using open_ssl:
```
4007B2B0CD7F0000:error:0A0000C7:SSL routines:tls_process_client_certificate:peer did not return a certificate:../ssl/statem/statem_srvr.c:3510:
```

### Collector version

0.101.0

## Environment
Kubernetes

### OpenTelemetry Collector configuration

```yaml
exporters:
    debug:
        verbosity: detailed
receivers:
    prometheus:
        config: {}
        target_allocator:
            collector_id: ${POD_NAME}
            endpoint: https://tls-debug-service:443
            interval: 30s
            tls:
                ca_file: /tls/ca.crt
                cert_file: /tls/tls.crt
                key_file: /tls/tls.key
service:
    pipelines:
        metrics:
            exporters:
                - debug
            processors: []
            receivers:
                - prometheus
```

### Log output

```shell
From the Collector:

2024-06-04T13:18:58.967Z info service@v0.101.0/service.go:102 Setting up own telemetry...
2024-06-04T13:18:58.967Z info service@v0.101.0/telemetry.go:103 Serving metrics {"address": ":8888", "level": "Normal"}
2024-06-04T13:18:58.967Z info exporter@v0.101.0/exporter.go:275 Development component. May change in the future. {"kind": "exporter", "data_type": "metrics", "name": "debug"}
2024-06-04T13:18:58.968Z info service@v0.101.0/service.go:169 Starting otelcol... {"Version": "0.101.0", "NumCPU": 12}
2024-06-04T13:18:58.968Z info extensions/extensions.go:34 Starting extensions...
2024-06-04T13:18:58.969Z info prometheusreceiver@v0.101.0/metrics_receiver.go:275 Starting discovery manager {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
2024-06-04T13:18:58.970Z info prometheusreceiver@v0.101.0/metrics_receiver.go:121 Starting target allocator discovery {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
2024-06-04T13:18:58.980Z info prometheusreceiver@v0.101.0/metrics_receiver.go:253 Scrape job added {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/default/nginx-servicemonitor/0"}
2024-06-04T13:18:58.980Z info service@v0.101.0/service.go:195 Everything is ready. Begin running and processing data.
2024-06-04T13:18:58.980Z warn localhostgate/featuregate.go:63 The default endpoints for all servers in components will change to use localhost instead of 0.0.0.0 in a future version. Use the feature gate to preview the new default. {"feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-06-04T13:18:58.980Z info prometheusreceiver@v0.101.0/metrics_receiver.go:340 Starting scrape manager {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
2024-06-04T13:18:58.995Z error refresh/refresh.go:71 Unable to refresh target groups {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "discovery": "http", "config": "serviceMonitor/default/nginx-servicemonitor/0", "err": "Get \"https://tls-debug-service:443/jobs/serviceMonitor%2Fdefault%2Fnginx-servicemonitor%2F0/targets?collector_id=collector-with-ta-collector-0\": remote error: tls: certificate required"}
github.com/prometheus/prometheus/discovery/refresh.(*Discovery).Run
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ItielOlenick**
**Description:** 
Translating relevant fields from TargetAllocator's confighttps to be used in the service discovery http client config
This assures that the relevant config defiend for the TA is used by the service discovery jobs in addition to the scrape configs (as solved by https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31449)

**Link to tracking Issue:** Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33370

**Testing:**
Added unit tests.

**Documentation:**

### CODE REVIEW

**Comment by ItielOlenick** [comment]
Keep alive

**Review by dashpole — APPROVED** [review]

**Comment by crobert-1** [comment]
From failing CI/CD action:
```
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.
```

**Comment by ItielOlenick** [comment]
@crobert-1 Ran make gotidy

**Comment by crobert-1** [comment]
_Added ready to merge as this was approved by a code owner who is also an approver._

**Comment by dmitryax** [comment]
@ItielOlenick can you please rebase to fix the conflict?

**Comment by ItielOlenick** [comment]
@dmitryax Done

**Comment by ItielOlenick** [comment]
@dmitryax @crobert-1 Can we get this merged? 
This PR introduces new imports, and by the time the GitHub checks get approval and pass there is a new conflict with the existing go.mod

File: .chloggen/httpSD-client-config-from-TA-confighttp.yaml
Comment (andrzej-stencel) [inline .chloggen/httpSD-client-config-from-TA-confighttp.yaml:10]: I feel the note does not reflect this change well. Would the following be a better description? Feel free to modify it, my understanding of this change is limited.

```suggestion
note: Ensure TLS config from Target Allocator is used in the receiver's service discovery
```

Comment (ItielOlenick) [reply .chloggen/httpSD-client-config-from-TA-confighttp.yaml:10]: TLS was the main driver for this PR but as [discussed](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33370#issuecomment-2214222263) we wanted to translate as much as possible from confighttp

Comment (ItielOlenick) [reply .chloggen/httpSD-client-config-from-TA-confighttp.yaml:10]: What about
```suggestion
note: Ensure Target Allocator's confighttp is used in the receiver's service discovery
```

Comment (swiatekm) [reply .chloggen/httpSD-client-config-from-TA-confighttp.yaml:10]: > What about

That looks clear enough to me.

**Review by swiatekm — APPROVED** [review]
LGTM

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**
